### PR TITLE
feat: add minAge, maxAge to birthday options (#561)

### PR DESCRIPTION
**Pull request by gerardolima**
implements (#561)

### CODE REVIEW

File: chance.js [hunk 1]
```diff
@@ -926,6 +927,22 @@
                 min: min,
                 max: max
             });
+        } if (options && (options.minAge || options.maxAge)) {
+            testRange(options.minAge > options.maxAge, "Chance: minAge cannot be greater than maxAge.");
+
+            var minAge = typeof options.minAge !== "undefined" ? options.minAge : 1;
+            var maxAge = typeof options.maxAge !== "undefined" ? options.maxAge : 100;
+
+            var minDate = new Date(now.getFullYear() - maxAge - 1, now.getMonth(), now.getDate());
```
Comment (victorquinn) [inline chance.js:936]: You create the variable `currentYear` above but then you recompute it here. Please re-use the variable you created on `918` rather than recalling `now.getFullYear()`

File: chance.js [hunk 2]
```diff
@@ -926,6 +927,22 @@
...
+            var maxDate = new Date(now.getFullYear() - minAge, now.getMonth(), now.getDate());
```
Comment (victorquinn) [inline chance.js:937]: I suspect you may have edge cases here when `minAge` is 0 where you could get a birthday in the future

For example if it is January 1, 2022, then the valid range of birthdays for 0 would be January 1, 2021-January 1, 2022

But I suspect here you'd get something in the range January 1, 2022 - December 31, 2022

I would suggest writing some unit tests to test these gnarly edge cases

File: chance.js [hunk 3]
```diff
@@ -926,6 +927,22 @@
                 min: min,
                 max: max
             });
+        } if (options && (options.minAge || options.maxAge)) {
```
Comment (victorquinn) [inline chance.js:930]: Should this be `else if`?

Seems like this is a dangling if statement...

**Review by victorquinn — CHANGES_REQUESTED** [review]
Overall I added a few comments, but this PR has zero unit tests.

Please write some tests for your codepath to ensure stability and expected results. I suggested some around odd edge cases, but even some more basic ones would be useful to add tests for the functionality you are trying to achieve here.

For example, if you specify `minAge`, and run it 1000 times, do you always get a birthday that is actually older than that value? Same with `maxAge` and then the combination of the two.

Without any tests I do not have confidence that it will function as expected.

Thank you for your contribution here and apologies on the delay reviewing it, I did not get notified when it got un-archived!

**Comment by gerardolima** [comment]
I'll work on these topics as soon as I can get some time.

File: chance.js (hunk 2 above)
Comment (gerardolima) [reply chance.js:937]: hey, @victorquinn, your comment relates to a doubt I have about tests in Chance. As `chance.birthday()` only calculates boundaries and call `chance.date()`, my first thought was _spying_ on date and assessing the arguments it actually receives. I saw no _mocks/stubs/spies_ on the test suite, though, so I wonder: would you be ok with this approach?

About this particular case, the boundaries are correct, as below:
```js
// test setup
const now=new Date(2022,0,1)
const minAge=0
const maxAge=0

// code copied from implementation
const currentYear = now.getFullYear()
let minDate = new Date(currentYear - maxAge - 1, now.getMonth(), now.getDate())
let maxDate = new Date(currentYear - minAge, now.getMonth(), now.getDate());

maxDate.setDate(maxDate.getDate() +1);
maxDate.setMilliseconds(maxDate.getMilliseconds() -1);

// assessment of the date limits
console.log(minDate) // 2021-01-01T00:00:00.000Z
console.log(maxDate) // 2022-01-01T23:59:59.999Z
```

**Comment by gerardolima** [comment]
hey, @victorquinn, I believe I had addressed the topics you raised. I'm looking forward for your feedback :)

File: chance.js (hunk 2 above)
Comment (gerardolima) [reply chance.js]: I just pushed a version with tests without mocks that were able to find some corner cases. Maybe this should not be so important in this context.

File: docs/person/birthday.md [hunk 4]
```diff
@@ -51,5 +51,13 @@ chance.birthday({ year: chance.year({ min: 1450, max: 1500 }) });
 => Fri Nov 26 1469 09:17:13 GMT-0500 (EST)
 ```
 
+Can specify maxAge and/or minAge. This will return a date which yields to an age beteween the given range. Attention on limits: the full day of birthdays are considered to be part of the allowed range (from the first milisecond of the minimun date to the last second of the maximun date).
```
Comment (victorquinn) [inline docs/person/birthday.md:54]:
```suggestion
Can specify maxAge and/or minAge. This will return a date which yields to an age between the given range. Attention on limits: the full day of birthdays are considered to be part of the allowed range (from the first millisecond of the minimum date to the last second of the maximum date).
```

**Review by victorquinn — APPROVED** [review]
Looks great! Some typos (for which I suggested changes) and a debug message I'd like you to remove before merging

Apologies for the delay, I maintain this library as a hobby on my nights and weekends outside of work.

**Comment by gerardolima** [comment]
hey, @victorquin, thank again you for your feedback! I just pushed the changes to fix the topics you raised. As these were just on comments, I think -- hope 😄 -- the PR is now ready to be merged.

**Comment by victorquinn** [comment]
Thanks again @gerardolima !

This is live on npm as `1.1.9`
